### PR TITLE
[FEATURE: UIPT-22] - Join us

### PR DESCRIPTION
**Pull request by TorskyM**

### CODE REVIEW

File: src/app/join-us/join-us.component.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    success: 'You have successfully subscribed to our newsletter',
+    subscribe: 'Subscribe',
+    unsubscribe: 'Unsubscribe',
+    message: 'Subscribe to Newsletter:'
+  };
+  private emailRegEx = /^(?=^.{6,25}$)(\D)+(\w)*((\.(\w)+)?)+@(\D)+(\w)*((\.(\D)+(\w)*)+)?(\.)[a-z]{2,4}$/;
+
+  constructor(private localStorageService: LocalStorageService){}
+
+  ngOnInit() {
```
Comment (AlexLarusso) [inline src/app/join-us/join-us.component.ts]: Typescript!
```
public ngOnInit(): void
```

File: src/app/local-storage.service.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class LocalStorageService {
+  public subscribed = false;
+  public userEmail = '';
```
Comment (AlexLarusso) [inline src/app/local-storage.service.ts]:
```
 public subscribed = false;
 public userEmail = '';
```
These props shouldn't be part of this service. But for now let it be, just left comment `TODO: move it form here`

File: src/app/local-storage.service.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  public onRefresh(item: string) {
+    if (localStorage.getItem(item) !== null) {
+     return this.subscribed = true;
```
Comment (AlexLarusso) [inline src/app/local-storage.service.ts]: Im not sure. Why you returning assignment?

Comment (TorskyM) [reply src/app/local-storage.service.ts]: This is required to render the relevant subscription notification message after the page is reloaded

File: src/app/shared/button/button.component.ts [hunk 4]
```diff
@@ -7,5 +7,6 @@ import { Component, Input } from '@angular/core';
 })
 
 export class ButtonComponent {
-  @Input() public buttonClass: string;
+  @Input() public buttonClass: string; 
+  @Input() public disabled: boolean;
```
Comment (AlexLarusso) [inline src/app/shared/button/button.component.ts]: Rename to isDisabled

**Pull request merged**